weatherrouting/routers/router.py

```diff
@@ -55,10 +56,15 @@ def __init__(self, time, path=[], isochrones=[], position=None, progress=0):
 class Router:
 	PARAMS = {}
 
-	def __init__ (self, polar, grib, pointValidity):
+	def __init__ (self, polar, grib, validity_func):
 		self.polar = polar
 		self.grib = grib
-		self.pointValidity = pointValidity
+		if len(inspect.signature(validity_func).parameters) == 2:
+			self.pointValidity = validity_func
+			self.lineValidity = None
+		elif len(inspect.signature(validity_func).parameters) == 4:
+			self.pointValidity = None
+			self.lineValidity = validity_func
```

Comment:
Default set lineValidity and pointValidity to None before the if-elif

Reply:
the provided pointValidity function could have many issues, could be implemented or not, and have mismatching params in type, number and position. I think that It would better to implement an headache preventing check but it have to be more comprehensive.